awx/main/tasks.py

```diff
@@ -1534,6 +1528,12 @@ def run(self, pk, **kwargs):
 
         try:
             self.post_run_hook(self.instance, status)
+        except PostRunError as exc:
+            if status == 'successful':
+                status = exc.status
+                extra_update_fields['job_explanation'] = exc.args[0]
+                if exc.tb:
+                    extra_update_fields['result_traceback'] = exc.tb
```

Comment:
@jladdjr I wanted to draw your attention to this.

I don't like modifying the logic inside of the main `run` method, but I can't think of any other way to do this which won't flip the status of the update back-and-fourth. In these cases, the `ansible-inventory` subprocess is fully successful, but for some reason, the saving process errors.